staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types_jsonschema.go

```diff
@@ -117,6 +117,19 @@ type JSONSchemaProps struct {
 	// Defaults to atomic for arrays.
 	// +optional
 	XListType *string `json:"x-kubernetes-list-type,omitempty" protobuf:"bytes,42,opt,name=xKubernetesListType"`
+
+	// x-kubernetes-map-type annotates a map to further describe its topology.
+	// This extension must only be used on maps and may have 2 possible values:
```

Comment:
would refer to the type instead of "map":

This extension must only be used when type is `object` and may have 2 possible values

Reply:
Right, on top of that, I would like to mention that we also want to support this extension on "go structures", which are also "object" types in openapi.